sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala

```diff
@@ -163,6 +187,7 @@ case class IsNotNull(attribute: String) extends Filter {
  */
 @Stable
 case class And(left: Filter, right: Filter) extends Filter {
+  override def sql: String = s"${left.sql} AND ${right.sql}"
```

Comment:
> for simple case like DELETE by filters in this pr, just pass the filter to datasource is more suitable, a 'spark job' is not needed.

BTW, I don't know why this is needed but it won't work with the combinations of `AND` and `OR`. `a OR b AND C`

Reply:
Thanks @HyukjinKwon . 
1. Why this function is introduced:
sources.Filter is used for datasource to filter data, but before that, datasource needs to understand the filters. Add 'sql' function here helps datasources like JDBC can rebuild the sql string to filter data.
2. For case like `a OR b AND c`, the final expected filter should be `OR(a, AND(b,c))`. The logic here indeed not well considered, including constants and parentheses. Will submit a fix soon.

Reply:
@HyukjinKwon , I'm doubt if I understand you correctly. Can you explain more about "it won't work with the combinations of `AND` and `OR`. `a OR b AND C`"?

Reply:
I meant the fix you just did https://github.com/apache/spark/pull/25115/commits/bc9daf9329868961bcd2d8cf71e882a26ddb5857

Reply:
Thanks @HyukjinKwon